components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1233,6 +1233,7 @@ bool RewardsServiceImpl::IsWalletCreated() {
 void RewardsServiceImpl::GetPublisherActivityFromUrl(uint64_t windowId,
                                                      const std::string& url,
                                                      const std::string& favicon_url) {
+  GetGrant(std::string(), std::string());
```

Comment:
We  should be fetching grant when you open the panel and not when we fetch publisher data from the url. This code will fetch grant for every site that you visit. Let's add check in JS when we mount app component.